File: Armageddon/src/main/java/de/tomjuri/armageddon/util/KeyBindUtil.java
package de.tomjuri.armageddon.util;

import de.tomjuri.armageddon.Armageddon;
import de.tomjuri.armageddon.mixin.MinecraftInvoker;
import net.minecraft.client.settings.KeyBinding;

public class KeyBindUtil {
    public static void leftClick() { ((MinecraftInvoker) Ref.mc()).invokeClickMouse(); }
    public static void rightClick() { ((MinecraftInvoker) Ref.mc()).invokeRightClickMouse(); }
    public static void setKey(int key, boolean pressed) { KeyBinding.setKeyBindState(key, pressed); }
}

File: src/main/kotlin/de/tomjuri/gemdigger/util/KeyBindUtil.kt
package de.tomjuri.gemdigger.util

import de.tomjuri.gemdigger.mixin.MinecraftInvoker
import net.minecraft.client.Minecraft
import net.minecraft.client.settings.KeyBinding

object KeyBindUtil {
    fun rightClick() { (mc as MinecraftInvoker).invokeRightClickMouse() }
    fun setKey(key: Int, heldDown: Boolean) { KeyBinding.setKeyBindState(key, heldDown) }
}